x-pack/plugins/observability/public/data_handler.ts

```diff
@@ -31,14 +35,26 @@ export function getDataHandler<T extends ObservabilityFetchDataPlugins>(appName:
   }
 }
 
-export async function fetchHasData(): Promise<Record<ObservabilityFetchDataPlugins, boolean>> {
-  const apps: ObservabilityFetchDataPlugins[] = ['apm', 'uptime', 'infra_logs', 'infra_metrics'];
+export async function fetchHasData(absoluteTime: {
+  start: number;
+  end: number;
+}): Promise<Record<ObservabilityFetchDataPlugins, HasDataResponse>> {
```

Comment:
@shahzad31 the `hasData` function should not receive any props (`absoluteTime`). The idea of hasData is to see if the user has ever set up an agent. If you do it and the user selects a time range where no data is available the chart section will be gone and instead the empty section will show up, and we don't want this kind of jump just because the user changed the time range.

Reply:
so right now it looks bit cluttered, in my next PR, there will be separate hook for each hasData call. in rum has data i also need most active service name in that time range, so it's a i don't want to make three es calls.

Reply:
date range is only passed to ux has data call and even there it only applies to service name aggs. it doesn't effect that has data call.

Reply:
And what will happen if you select a date which doesn't have any data? Will hasData return false?

Reply:
nope, it will only show core vitals in empty state.

Reply:
in other words, it will not return serviceName, but it will return hasData: true

Reply:
this is how query looks like for rum has data call

```
GET apm-*/_search
{
  "size": 0,
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "transaction.type": "page-load"
          }
        }
      ]
    }
  },
  "aggs": {
    "services": {
      "filter": {
        "range": {
          "@timestamp": {
            "gte": 1601632965783,
            "lte": 1601633865783,
            "format": "epoch_millis"
          }
        }
      },
      "aggs": {
        "mostTraffic": {
          "terms": {
            "field": "service.name",
            "size": 1
          }
        }
      }
    }
  }
}
```